### PR TITLE
[MCIV-969] Integrate Update Profile for Re-Identification

### PR DESCRIPTION
**Pull request by sfxvchauhan**
### Jira Tickets :gem:
 
- [MCIV-969](https://seamfix.atlassian.net/browse/MCIV-969)
 
### Details :memo:
 
- Integrate Update Profile for Re-Identification
 
### Related :mag:
 
- Links to Pull Requests or other issues related to this work
 
### Screenshots / Demos :camera:
 
test-case

[MCIV-969]: https://seamfix.atlassian.net/browse/MCIV-969?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/update-profile/update-profile.service.ts [hunk 1]
```diff
@@ -96,88 +97,62 @@ export class UpdateProfileService implements IUpdateProfile {
...
-                                                            "EXTERNAL_REFERENCE": "294420478", // TODO
-                                                            "EXTERNAL_SYSTEMS_LOG_REFERNCE": dto.objectId,
-                                                            "EXT_SUBSCRIBER_CODE": "11946322", // TODO
-                                                            "UPDT_ONLY_SERVICE": "N", // TODO
-                                                            "ACCOUNT_LINK_CODE": "725093863" // TODO
+                                                        "REQUEST": {
+                                                           "_attributes": { 
+                                                                "CLIENT_ID": "MTNCI",
+                                                                "ENTITY_ID": dto.msisdn,
+                                                                "EXTERNAL_APPLICATION": "axonext",
```
Comment (sfxeezeakacha) [inline src/update-profile/update-profile.service.ts]: Please pass this from the parent app.

File: src/update-profile/update-profile.service.ts [hunk 2]
```diff
@@ -96,88 +97,62 @@ export class UpdateProfileService implements IUpdateProfile {
...
+                                                                "EXTERNAL_SYSTEMS_LOG_REFERNCE": dto.externalLogReference,
+                                                                "EXTERNAL_USER": "axonext",
```
Comment (sfxeezeakacha) [inline src/update-profile/update-profile.service.ts]: Please pass this from the parent app.

File: src/update-profile/update-profile.service.ts [hunk 3]
```diff
@@ -96,88 +97,62 @@ export class UpdateProfileService implements IUpdateProfile {
...
+                                                                "INFO_LEVEL": "3",
```
Comment (sfxeezeakacha) [inline src/update-profile/update-profile.service.ts]: Please pass this from the parent app.

File: src/update-profile/update-profile.service.ts [hunk 4]
```diff
@@ -415,4 +391,19 @@ export class UpdateProfileService implements IUpdateProfile {
 			return jsonObj;
 		}
 	}
+
+    getGender(gender: string): string {
+        if(!gender) {
+            return ''
+        } else {
+            if(gender.toLocaleLowerCase() === 'male') {
```
Comment (sfxeezeakacha) [inline src/update-profile/update-profile.service.ts]: Please make the `male` and the `female` texts constants.

Comment (sfxvchauhan) [reply src/update-profile/update-profile.service.ts]: Done.

File: src/update-profile/update-profile.service.ts (hunk 1 above)
Comment (sfxvchauhan) [reply src/update-profile/update-profile.service.ts]: Done.

File: src/update-profile/update-profile.service.ts (hunk 2 above)
Comment (sfxvchauhan) [reply src/update-profile/update-profile.service.ts]: Done.

File: src/update-profile/update-profile.service.ts (hunk 3 above)
Comment (sfxvchauhan) [reply src/update-profile/update-profile.service.ts]: Done.

**Review by sfxeezeakacha — APPROVED** [review]

**Review by sfxkalli — APPROVED** [review]

**Pull request merged**